web/src/main/webapp/my-app/marketplace/controllers.js

```diff
@@ -84,34 +84,42 @@ define(['angular', 'jquery', 'require'], function(angular, $, require) {
 
       $scope.addToHome = function addToHome(portlet) {
         var fname = portlet.fname;
-        var ret = layoutService.addToHome(portlet);
-        ret.success(function(request, text) {
-          angular.element('.fname-'+fname)
+        var ret = layoutService.addToHome(portlet, $sessionStorage.layout);
+        ret.then(
+          function successCallback(response) {
+
+            $log.log('Added ' + portlet.fname + ' successfully');
+            angular.element('.fname-'+fname)
             .html('<i class="fa fa-check"></i> Added Successfully')
             .prop('disabled', true)
             .removeClass('btn-add')
             .addClass('btn-added');
-          $scope.$apply(function() {
```

Comment:
Per AngularJS documentation: Events such $http callbacks are all wrapped in $scope.$apply(). So there's no need to call it again. Calling $apply inside $apply throws an error.